airflow/www/views.py

```diff
@@ -911,10 +928,11 @@ def index(self):
                 dag.can_trigger = dag.can_edit and can_create_dag_run
                 dag.can_delete = get_airflow_app().appbuilder.sm.can_delete_dag(dag.dag_id, g.user)
 
-            dagtags = session.execute(select(func.distinct(DagTag.name)).order_by(DagTag.name)).all()
+            dag_tags = session.execute(select(func.distinct(DagTag.name)).order_by(DagTag.name)).all()
+            tags_operator = ["and", "or", "not"]
             tags = [
-                {"name": name, "selected": bool(arg_tags_filter and name in arg_tags_filter)}
-                for name, in dagtags
+                {"name": name, "selected": [op for op in tags_operator if f"{name}|{op}" in arg_tags_filter]}
```

Comment:
I don't like how this will make every single tag appear 3 times. If someone has many dags, this can quickly become unusable.